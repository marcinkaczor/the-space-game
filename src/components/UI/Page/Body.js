import React from 'react'
import PropTypes from 'prop-types'
import { Box } from '@material-ui/core'

const PageBody = ({ children, column }) => (
  <Box
    flex={1}
    alignSelf='center'
    display='flex'
    alignItems='center'
    flexDirection={column && 'column'}
    justifyContent={column && 'center'}
  >
    {children}
  </Box>
)

PageBody.propTypes = {
  children: PropTypes.node.isRequired
}

export default PageBody
