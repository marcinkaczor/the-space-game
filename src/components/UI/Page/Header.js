import React from 'react'
import PropTypes from 'prop-types'
import { Typography } from '@material-ui/core'

const PageHeader = ({ children, ...props }) => (
  <Typography {...props}>{children}</Typography>
)

PageHeader.propTypes = {
  children: PropTypes.node.isRequired
}

export default PageHeader
