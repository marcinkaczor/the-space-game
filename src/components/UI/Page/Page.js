import React from 'react'
import PropTypes from 'prop-types'
import { Container, Box } from '@material-ui/core'

const Page = ({ children }) => (
  <Container maxWidth='sm'>
    <Box display='flex' flexDirection='column' height='100vh'>
      {children}
    </Box>
  </Container>
)

Page.propTypes = {
  children: PropTypes.node.isRequired
}

export default Page
