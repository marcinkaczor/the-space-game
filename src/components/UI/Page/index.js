import Page from './Page'

import Header from './Header/Header'
import Body from './Body/Body'

export { Header as PageHeader, Body as PageBody }

export default Page
