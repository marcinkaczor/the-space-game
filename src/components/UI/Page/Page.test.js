/* global describe, test, it, expect */
import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import Page from '.'

describe('Page', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<Page><div>Hello</div></Page>)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders with child', () => {
    const Child = () => <div>Hello from child</div>

    const wrapper = mount(
      <Page>
        <Child />
      </Page>
    )
    expect(wrapper.find(Child).length).toEqual(1)
  })
})
