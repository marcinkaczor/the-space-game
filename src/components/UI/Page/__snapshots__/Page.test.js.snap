// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page snapshot renders 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthSm"
>
  <div
    className="MuiBox-root MuiBox-root-10"
  >
    <div>
      Hello
    </div>
  </div>
</div>
`;
